r/geoarrow/R/vctr.R

```diff
@@ -0,0 +1,66 @@
+
+as_geoarrow_vctr <- function(x, ..., schema = NULL) {
+  if (inherits(x, "geoarrow_vctr") && is.null(schema)) {
+    return(x)
+  }
+
+  stream <- as_geoarrow_array_stream(x, ..., schema = schema)
+  chunks <- nanoarrow::collect_array_stream(stream, validate = FALSE)
+  new_geoarrow_vctr(chunks, stream$get_schema())
+}
+
+new_geoarrow_vctr <- function(chunks, schema) {
+  offsets <- .Call(geoarrow_c_vctr_chunk_offsets, chunks)
+  structure(
+    seq_len(offsets[length(offsets)]),
+    schema = schema,
+    chunks = chunks,
+    chunk_offsets = offsets,
+    class = c("geoarrow_vctr", "wk_vctr")
+  )
```

Comment:
Could we store just the array stream and wk attributes (crs, geodesic)? I think we can retrieve schema and offsets from the array stream when they're required?

Reply:
> For crs and geodesic I'd envisioned keeping those with the schema (i.e., modifying them via wk_set_crs() or wk_set_geodesic() would modify the schema instead of set an attribute on the vctr). Also not sure I thought that all the way through or not!

That makes sense. I'd forgotten these are available in the schema.

> I'm pretty sure we need list() of nanoarrow_array here and offsets so that we can efficiently subset (see the bit I added). Once you have a stream, you don't know how many chunks or elements there are and you loose random access. I definitely may have missed something though!

Ok.